test_rclcpp/test/test_allocator.cpp

```diff
@@ -255,18 +255,18 @@ struct InstrumentedAllocator : public allocator_pointer_traits<T>
 
   // construct and destroy should not be required to implement allocator_traits,
   // this is a bug in gcc 4.8 for usage in std::pair, std::map
-  template<typename _Up, typename ... _Args>
-  void
-  construct(_Up * __p, _Args && ... __args)
+  template<typename U, typename ... Args>
+  typename std::enable_if<!std::is_const<U>::value>::type
+  construct(U * ptr, Args && ... args)
   {
-    ::new(static_cast<void *>(__p))_Up(std::forward<_Args>(__args) ...);
+    ::new(reinterpret_cast<void *>(ptr))U(std::forward<Args>(args) ...);
```

Comment:
That was leftover from experimenting with what would make it compile. Actually, no cast is needed because it will implicitly cast from the U pointer type to `void*`.